### PR TITLE
USSD registration validate day and date

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
Affecting current states 6 and 14.
Add additional state for invalid date.

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

File: src/ussd_registration.js [hunk 1]
```diff
@@ -329,6 +330,47 @@ go.app = function() {
...
+            var monthAndYear;
+            var day;
+            if (self.im.user.answers.state_last_period_month) {     // flow via st-05 & st-06
+                monthAndYear = self.im.user.answers.state_last_period_month;
+                day = self.im.user.answers.state_last_period_day;
+            }
+            else {                                                  // flow via st-12 & st-13
+                monthAndYear = self.im.user.answers.state_baby_birth_month_year;
+                day = self.im.user.answers.state_baby_birth_day;
+            }
```
Comment (gsvr) [inline src/ussd_registration.js]: Here's a Javascripty shorter way of doing the above:

```
var monthAndYear = self.im.user.answers.state_last_period_month ||
                   self.im.user.answers.state_baby_birth_month_year;
var day = self.im.user.answers.state_last_period_day ||
          self.im.user.answers.state_baby_birth_day;
```

File: src/ussd_registration.js [hunk 2]
```diff
@@ -124,6 +124,7 @@ go.app = function() {
     // START STATE
 
         self.add('state_start', function(name) {
+            self.im.user.answers = {};
```
Comment (gsvr) [inline src/ussd_registration.js:127]: We need a test to see that this is working. There's an example test in go-ndoh nurse_ussd.

**Comment by gsvr** [comment]
@NicolaasDieBaas Done for now.

File: src/ussd_registration.js (hunk 1 above)
Comment (NicolaasDieBaas) [inline src/ussd_registration.js]: nice!

**Comment by NicolaasDieBaas** [comment]
@gsvr Hope it's good now :)

**Comment by gsvr** [comment]
@NicolaasDieBaas Please run `npm test` again - I don't think the compiled files were updated for the comment removal. :+1: when that's done.

**Pull request merged**